### ISSUE TITLE
Function timeout undefined when running in shell with debugger

### ISSUE THREAD

**Issue by AlexLavoie42** [firebase/firebase-tools#5573]
### [REQUIRED] Environment info

**firebase-tools:** 11.24.0

**Platform:** Arch

### [REQUIRED] Test case
Any long running function should work.
```js
exports.test = functions.runWith({timeoutSeconds: 120}).https.onCall(async (data, context) => {
  await new Promise((res, rej) => setTimeout(res, 60000));
  return "Done";
})
```

### [REQUIRED] Steps to reproduce

Run a function using functions:shell with a debugger attached.
```firebase functions:shell --inspect-functions```

### [REQUIRED] Expected behavior

Function should not timeout with debugger attached to avoid timeout when inspecting a breakpoint. Should at the very least be what is set in ```timeoutSeconds```

### [REQUIRED] Actual behavior

Function times out shortly after running with the following message:
```
i  Your function timed out after ~undefineds. To configure this timeout, see
      https://firebase.google.com/docs/functions/manage-functions#set_timeout_and_memory_allocation.
i  Request to function failed: Error: socket hang up
```

**Comment by blidd-google** [comment]
Hi @AlexLavoie42, unfortunately I am unable to reproduce your issue — functions timeouts look like they are properly disabled when a debugger is attached. Because the output log is saying the function timed out after "~undefineds" seconds, my guess is that the function timeout is actually properly disabled and a timeout is occurring for another reason. Could you provide more context about what you are trying to test?

**Comment by AlexLavoie42** [comment]
Also just to note, by downgrading to v11.19 I was able to temporarily get around the issue

**Comment by blidd-google** [comment]
Thanks for the additional context — I was able to replicate your issue once I upgraded my node version from 16 to 19. Unfortunately, firebase functions does not currently support node version 19 (officially, only versions 10 / 12 / 14 / 16 are supported); if at all possible, downgrading to v16 will resolve your issue in the meantime. 

In any case, it's good to know that upgrading the node engine to v19 will cause a breaking change to the way we disable timeouts in the emulator shell. I'll escalate this issue to the team and see if we can implement a workaround until we have time to fully support node engines >16.
